compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2065,6 +2065,236 @@ public void exitStringTemplateLiteral(BallerinaParser.StringTemplateLiteralConte
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startSelectClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitSelectClause(BallerinaParser.SelectClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        boolean isSelectAll = ctx.MUL() != null;
+        boolean isGroupByClauseAvailable = ctx.groupByClause() != null;
+        boolean isHavingClauseAvailable = ctx.havingClause() != null;
+        this.pkgBuilder.endSelectClauseNode(isSelectAll, isGroupByClauseAvailable, isHavingClauseAvailable,
+                getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterSelectExpressionList(BallerinaParser.SelectExpressionListContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startSelectExpressionList();
+    }
+
+    @Override
+    public void exitSelectExpressionList(BallerinaParser.SelectExpressionListContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endSelectExpressionList(getWS(ctx), ctx.getChildCount() / 2 + 1);
+    }
+
+    @Override
+    public void enterWhereClause(BallerinaParser.WhereClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startWhereClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitWhereClause(BallerinaParser.WhereClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endWhereClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterWindowClause(BallerinaParser.WindowClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startWindowClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitWindowClause(BallerinaParser.WindowClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endWindowsClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterStreamingInput(BallerinaParser.StreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startStreamingInputNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitStreamingInput(BallerinaParser.StreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        boolean isWindowAvailable;
+        boolean isFirstWhereClauseAvailable;
+        boolean isSecondWhereClauseAvailable;
+        isWindowAvailable = ctx.windowClause() != null;
+        isFirstWhereClauseAvailable = ctx.whereClause(0) != null;
+        isSecondWhereClauseAvailable = ctx.whereClause(1) != null;
```

Comment:
Move variable declaration and assignment to a single line for all the above 3 variables.

Reply:
fixed.